sentry-log/src/integration.rs

```diff
@@ -37,13 +38,27 @@ impl Integration for LogIntegration {
         cfg.extra_border_frames
             .push("<sentry_log::Logger as log::Log>::log");
 
-        let filter = self.effective_global_filter();
+        let filters = self.create_filters();
+        let filter = filters.effective_global_filter();
         if filter > log::max_level() {
             log::set_max_level(filter);
         }
 
-        INIT.call_once(|| {
-            log::set_boxed_logger(Box::new(Logger::default())).ok();
+        INIT.call_once(move || {
+            // NOTE on safety:
+            // The way the current log-integration code is structured, we have
+            // no way to move the `dest_logger` from the integration to our own
+            // log instance without breaking the API.
+            // The `Once` here makes sure that we only ever do this unsafe
+            // `take` once, and there are no other pieces of code that read from
+            // from this integration instance.
+            let dest_log = unsafe {
+                let const_ptr = &self.dest_log as *const Option<Box<dyn log::Log>>;
+                let mut_ptr = const_ptr as *mut Option<Box<dyn log::Log>>;
+                (&mut *mut_ptr).take()
```

Comment:
Okay, I know that I may have suggested this in a meeting, but we need to find another way to do this. If we end up breaking all integration interfaces to make them builders, I guess that is fine and a good lesson on why we should not expose fields publicly :) 

To quote https://doc.rust-lang.org/nightly/nomicon/transmutes.html (UB = undefined behavior):

> - Transmuting an & to &mut is _always_ UB.
> - No you can't do it.
> - No you're not special.